pkg/machines/app.jsx

```diff
@@ -128,6 +134,42 @@ class App extends React.Component {
 
         const pathVms = path.length == 0 || (path.length > 0 && path[0] == 'vms');
 
+        let vmContent;
+        if (path.length > 0 && path[0] == 'vm') {
+            const vm = vms.find(vm => vm.name == cockpit.location.options.name && vm.connectionName == cockpit.location.options.connection);
+            if (!vm)
+                return null;
+
+            const connectionName = vm.connectionName;
+            const expandedContent = vm.isUi ? undefined : (
+                <VmExpandedContent vm={vm} vms={vms} config={config}
+                    libvirtVersion={systemInfo.libvirtVersion}
+                    notifications={this.state.resourceHasError[vm.id]
+                        ? Object.keys(this.state.notifications)
+                                .map(notificationId => this.state.notifications[notificationId])
+                                .filter(notification => notification.resourceId == vm.id)
+                                .map(notification => {
+                                    return (
+                                        <Alert variant='danger' key={notification.index}
+                                               isInline
+                                               onDismiss={() => this.onDismissErrorNotification(notification.index)}
+                                               title={notification.text}>{notification.detail}</Alert>
+                                    );
+                                }) : undefined}
```

Comment:
I think this matches the `?` on line 148. It's hard to read, can you indent it properly and also start the line with `:` so that it lines up with the '?'